### PR TITLE
Bugfix dl4j modules dependencies

### PR DESCRIPTION
**Pull request by sshepel**
## What changes were proposed in this pull request?

Fix dependencies for all dl4j to have a clean CUDA backend builds.

_Not sure about fixes for gym, jumpy and pydl4j, must be more cleaner way to fix build._

## How was this patch tested?

Tested locally.

## Quick checklist

The following checklist helps ensure your PR is complete:

- [ ] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [ ] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [ ] Created tests for any significant new code additions.
- [ ] Relevant tests for your changes are passing.

### CODE REVIEW

File: nd4j/nd4j-shade/protobuf/pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <parent>
         <artifactId>nd4j-shade</artifactId>
         <groupId>org.nd4j</groupId>
-        <version>1.0.0-SNAPSHOT</version>
+        <version>1.0.0-beta5</version>
```
Comment (AlexDBlack) [inline nd4j/nd4j-shade/protobuf/pom.xml]: This needs to be fixed... previous version was fine.

**Comment by saudet** [comment]
Looks fine, thanks! I don't think gym, jumpy, or pydl4j needs to depend on nd4j-native by default, so we should be able to remove that like it is for all the others modules.

**Comment by sshepel** [comment]
@saudet well I don't mind, but reset of the modules have nd4j backend as dependency for tests, in case of jympy and rest such kind of modules, I haven't seen the scope tag, that is why duplicated the same approach with properties that has been used for gym, with small changes for cuda...

**Comment by sshepel** [comment]
Going to open this PR against another repo...

**Pull request closed without merging**